### PR TITLE
Feedback Report: Feedback Badges

### PR DESCRIPTION
**Pull request by emcelroy**
PT Story: https://www.pivotaltracker.com/story/show/176162188

[#176162188]

### CODE REVIEW

File: js/components/portal-dashboard/feedback/activity-feedback-textarea.tsx [hunk 1]
```diff
@@ -29,23 +30,30 @@ export const ActivityFeedbackTextarea: React.FC<IProps> = (props) => {
 
   const updateFeedback = () => {
     if (activityId && studentId && updateActivityFeedback && textareaRef.current?.value !== undefined) {
+      const hasBeenReviewed = textareaRef.current.value !== "" ? true : false;
```
Comment (mklewandowski) [inline js/components/portal-dashboard/feedback/activity-feedback-textarea.tsx:33]: I was wondering about this case where the user deletes the feedback.  Good catch!

File: js/components/portal-dashboard/feedback/feedback-question-rows.tsx [hunk 2]
```diff
@@ -22,6 +23,17 @@ interface IProps {
 export const FeedbackQuestionRows: React.FC<IProps> = (props) => {
   const { answers, feedbacks, currentActivity, currentStudentId, students, updateQuestionFeedback } = props;
 
+  const getFeedbackIcon = (feedback: string, feedbackData: Map<string, any>, answer: Map<string, any>) => {
+    let feedbackBadge = <AwaitingFeedbackQuestionBadgeIcon />;
+    if (feedback !== "") {
+      feedbackBadge = feedbackValidForAnswer(feedbackData, answer)
+                      ? <GivenFeedbackQuestionBadgeIcon />
+                      : <UpdateFeedbackQuestionBadgeIcon />;
+    }
```
Comment (mklewandowski) [inline js/components/portal-dashboard/feedback/feedback-question-rows.tsx:32]: I probably would have used two ternary operators, but I think this is clearer!  All this functional programming has tried to beat the if/else out of me!

File: js/components/portal-dashboard/feedback/question-feedback-textarea.tsx [hunk 3]
```diff
@@ -31,7 +31,7 @@ export const QuestionFeedbackTextarea: React.FC<IProps> = (props) => {
   const updateFeedback = () => {
     if (answerId && textareaRef.current?.value !== undefined) {
       updateQuestionFeedback(answerId, {feedback: textareaRef.current?.value,
-                                        hasBeenReviewedForAnswerHash: answerHash(props.answer)});
+                                        hasBeenReviewedForAnswerHash: answerHash(answer)});
```
Comment (mklewandowski) [inline js/components/portal-dashboard/feedback/question-feedback-textarea.tsx:34]: Thanks for cleaning this up (props order, pulling answer from props)!  Sorry for rushing on this earlier today!

Comment (emcelroy) [reply js/components/portal-dashboard/feedback/question-feedback-textarea.tsx:34]: No problem at all 👍

**Review by mklewandowski — APPROVED** [review]
👍 LGTM!

**Review by eireland — APPROVED** [review]
Looks good to me!

**Pull request merged**